### PR TITLE
Nextcloud Caddyfile doesn't allow to use public sharing by link

### PR DESCRIPTION
**Pull request by pztrn**
This PR fixes it.

Without these changes filelist interface is loaded, but files list itself is missing.

### CODE REVIEW

File: nextcloud/Caddyfile [hunk 1]
```diff
@@ -28,6 +28,11 @@ my-nextcloud-site.com {
 		to /remote.php/{1}/{2}
 	}
 
+	rewrite {
+		r ^/public.php/(.+?)(\/?)$
+		to /public.php/(.+?)(\/?)$
```
Comment (mholt) [inline nextcloud/Caddyfile]: How does this work, exactly? I've never seen a rewrite to the same URI...

Comment (pztrn) [inline nextcloud/Caddyfile]: It perform requests to public.php when someone shared something by link. Confirming it on 11.0.3

See screenshot.

Comment (mholt) [inline nextcloud/Caddyfile]: That's bizarre. Any idea _why_ it works?

**Comment by francislavoie** [comment]
FWIW, I use owncloud and here's what works for me:

```
https://<domain> {
	root /var/www/html

	# PHP-FPM
	fastcgi / owncloud:9000 php {
		index index.php
	}

	rewrite {
		r /index.php/.*
		to /index.php?{query}
	}

	# client support (e.g. os x calendar / contacts)
	redir /.well-known/carddav /remote.php/carddav 301
	redir /.well-known/caldav /remote.php/caldav 301

	status 403 /forbidden

	# remove trailing / on some paths
	rewrite {
		r ^/remote.php/(webdav|caldav|carddav)(\/?)$
		to /remote.php/{1}
	}

	rewrite {
		r ^/remote.php/(webdav|caldav|carddav)/(.+)(\/?)$
		to /remote.php/{1}/{2}
	}

        rewrite {
                r ^/public.php/(dav|webdav|caldav|carddav)(\/?)$
                to /public.php/{1}
        }

        rewrite {
                r ^/public.php/(dav|webdav|caldav|carddav)/(.+)(\/?)$
                to /public.php/{1}/{2}
        }

	rewrite {
		r ^/(?:\.htaccess|data|config|db_structure\.xml|README)
		to /forbidden
	}

	header / {
		Strict-Transport-Security "max-age=31536000;"
	}

	gzip
	log /var/log/caddy/access.log
	errors /var/log/caddy/error.log

	tls <email>
}
```

File: nextcloud/Caddyfile (hunk 1 above)
Comment (pztrn) [inline nextcloud/Caddyfile]: Take a look to the rest of Caddyfile - it doing the VERY same thing. I've only fixed it to get public sharing by link working. Nothing else.

About why it works - well, IDK. It gives back 404 for me without this rewrite.

**Pull request merged**